### ISSUE TITLE
Computational differences between docker versions

### ISSUE THREAD

**Issue by RichardHooijmaijers** [docker/for-win#10167]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:
  - [x] I am using the Docker WSL2 Backend
  - [ ] I am using the Docker Hyper-V Backend

### Actual behavior

Computational results obtained with docker desktop deviate from results obtained with docker toolbox. 

### Expected behavior

No differences in outcome of calculations

### Information

I have performed calculations with a docker image (that uses fortran on the background) within a windows 10 environment running docker toolbox. When switching to docker desktop I noticed that some of the calculation results deviate from the results obtained with docker toolbox. After additional testing I saw that results on linux (inside and outside docker) are equal to the results on docker dekstop but not on docker toolbox.

I know docker toolbox is deprecated and that docker desktop gives me the 'correct' results. However I would like to understand how it is possible that results differ while the same image is used. Also I want to make sure that there is no possibility that differences will occur in the future when docker desktop is updated.

I have included some additional information below, any help with this issue is greatly appreciated.

  - Windows Version: 10 pro (build 19041.572) 
  - Docker Desktop Version: 3.1.0
  - WSL2 backend  
  - No virtualized Windows 
  - docker image running ubuntu 20.04/gfortran 9.3.0

**Comment by stephen-turner** [comment]
I can't immediately explain this. I agree that they should give the same result. What does the program do?

**Comment by RichardHooijmaijers** [comment]
The program fits non-linear mixed effects models. In order to do so it uses fortran on the background. There are various estimation methods in the software but the one I tested is fully reproducible on other systems

**Comment by stephen-turner** [comment]
I'm a bit stuck here. I'm trying to think what the differences could be. The only thing I can think of is that the Docker Machine VM might be running an older version of the Linux kernel. It seems unlikely that would make a difference, but it's possible.

Are there any files shared from the host into the containers and used in the calculations?

**Comment by RichardHooijmaijers** [comment]
There are files shared from the host. In order to perform the calculations, a model definition file and a dataset is copied from the host. 

Do you think that a difference in Linux kernel version could have an effect on the way fortran does it's job? Or does it effect the way resources are assigned (I honestly have no idea)?

**Comment by stephen-turner** [comment]
I was thinking it could possibly make a difference if there was a subtle non-deterministic behaviour in the code. For example if the result depended on the order in which two calculations are done but they're on separate threads. But I realise I'm clutching at straws a bit.

**Comment by RichardHooijmaijers** [comment]
I think that the order might be of importance here but I believe everything is single threaded in this situation. It might be a long shot but do you think differences in runtime libraries might be an issue here (e.g. see [link](https://stackoverflow.com/questions/58924506/different-results-for-log10-in-fortran-on-windows-and-linux))?

**Comment by stephen-turner** [comment]
No, because the container should encapsulate the libraries. But I'm wondering about 32-bit vs 64-bit operating systems.

**Comment by RichardHooijmaijers** [comment]
OK, that's a good point. I assumed that everything was just 64-bit. Is this maybe not the case for anything in docker toolbox?

**Comment by stephen-turner** [comment]
Actually, you're right, I think toolbox was all 64 bit, so that doesn't explain it either. Hmmm.

**Comment by stephen-turner** [comment]
OK, shall do, thanks.
